### PR TITLE
absolufy-imports

### LINKED ISSUE #5889: [Style] Absolute vs Relative imports
We're using a lot of relative imports in this package and we're not very consistent about when and how to use which. Counting occurrences in the entire code base slightly favours absolute imports (about 56%).

[PEP8](https://www.python.org/dev/peps/pep-0008/) recommends the usage of absolute imports over relative imports unless a complex package layout would require unnecessarily verbose import statements

> Absolute imports are recommended, as they are usually more readable and tend to be better behaved (or at least give better error messages) if the import system is incorrectly configured (such as when a directory inside a package ends up on sys.path):
> [...]
> However, explicit relative imports are an acceptable alternative to absolute imports, especially when dealing with complex package layouts where using absolute imports would be unnecessarily verbose:

I personally prefer absolute imports since they are much less ambiguous and portable. 

For instance, there are 24 `from .core import` and four `from ..core import` statements and we have six `core.py` files. It's not hard to figure out where the imports are pointing to but an absolute version removes ambiguity, especially for multi-level relative references

If we go for absolute imports, a follow up question would be whether we want to automate this, e.g. https://github.com/MarcoGorelli/absolufy-imports

Thoughts?

### PR DESCRIPTION
**Pull request by MarcoGorelli**
- [ ] Closes #5889
- [ ] Tests added / passed
- [ ] Passes `pre-commit run --all-files`

### CODE REVIEW

**Comment by GPUtester** [comment]
Can one of the admins verify this patch?

File: distributed/dashboard/components/shared.py [hunk 1]
```diff
@@ -21,14 +21,14 @@
...
 
-from distributed import profile
-from distributed.compatibility import WINDOWS
-from distributed.dashboard.components import DashboardComponent
-from distributed.dashboard.utils import update
-from distributed.utils import log_errors
+from ... import profile
+from ...compatibility import WINDOWS
+from ...utils import log_errors
+from ..utils import update
```
Comment (fjetter) [inline distributed/dashboard/components/shared.py:27]: This is a nice example how things can become quite ambiguous

**Review by fjetter — COMMENTED** [review]
Thanks for opening this. I think this is very helpful to see examples of how this would look like. 

I am still concerned about the ambiguity around duplicate module names like `utils` or `core`.

File: distributed/dashboard/components/shared.py (hunk 1 above)
Comment (gjoseph92) [reply distributed/dashboard/components/shared.py:27]: Agreed, this is just the sort of thing I'd like to avoid having to read.

**Comment by crusaderky** [comment]
should we just respect PEP4 and do everything absolute? I had not realised there were this many inconsistencies.

**Pull request closed without merging**